### PR TITLE
Add login documentation

### PR DESCRIPTION
**Pull request by cooncesean**
## Overview

This PR adds documentation around the "login" endpoint. There are a few questions that around the endpoint that will be asked inline.

### CODE REVIEW

File: auth-login.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  "password": "abc123",
+  "appId": "guidebook",
+  "appSecret": "8pp-s3cr3t-k3y"
+}
+```
+#### Responses
+
+**Success Response:**
+
+Status Code: `200`
```
Comment (cooncesean) [inline auth-login.md:33]: I assumed that a [200 status code](https://httpstatuses.com/200) was returned here, is that correct?

File: auth-login.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+```json
+{
+  "aboId": "123456",
```
Comment (cooncesean) [inline auth-login.md:37]: Is this identifier 100% unique for each user regardless of country code?

File: auth-login.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      "value": "f001382e-a150-4143-9336-81f75753f97b",
+      "expiration": "2016-02-04 05:09:57"
+    }
+  }
+}
+```
+
+**Failure Response:**
+
+Status Code: `400`
```
Comment (cooncesean) [inline auth-login.md:57]: I assumed that a [400 status code](https://httpstatuses.com/400) was returned here, is that correct?

File: auth-login.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+```json
+{
+  "abo_id": "123456",
```
Comment (cooncesean) [inline auth-login.md:61]: Is there a reason that `aboId` (camelCase) is the field name in the success response but `abo_id` (snake_case) is the field name in the failure response or was this a typo?

File: auth-login.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  "message": "failure"
```
Comment (cooncesean) [inline auth-login.md:62]: Are there other types of failure messages that will be returned here? eg: "Incorrect password.", "Username does not exist in the system.", etc.

File: auth-login.md (hunk 4 above)
Comment (cooncesean) [inline auth-login.md:61]: Would the failure response still return a valid `abo_id`? if the user could not authenticate themselves via their country code, username and password, it seems unlikely that DFX would return a valid `abo_id` in the response message.

File: auth-login.md (hunk 1 above)
Comment (tham-fookleong) [inline auth-login.md:33]: Correct.

File: auth-login.md (hunk 2 above)
Comment (tham-fookleong) [inline auth-login.md:37]: It unique for market and is not 100% unique in the region therefore you will need to include the country code.

File: auth-login.md (hunk 3 above)
Comment (tham-fookleong) [inline auth-login.md:57]: Correct.

File: auth-login.md (hunk 4 above)
Comment (tham-fookleong) [inline auth-login.md:61]: I will let Chin Han answer this.

File: auth-login.md (hunk 5 above)
Comment (tham-fookleong) [inline auth-login.md:62]: No there are no other failure types.

File: auth-login.md (hunk 4 above)
Comment (cooncesean) [inline auth-login.md:61]: re: camelCase `aboId`:

> Chin Han : It was a typo on my part sorry. I have fixed the documentation.

File: auth-login.md (hunk 2 above)
Comment (cooncesean) [inline auth-login.md:37]: Thx. Noted in 17e321f547cc74557a519a8fcc8b8e87574e4edf

**Pull request merged**